src/main/java/core/basesyntax/JavaDateTimeApi.java

```diff
@@ -96,23 +127,31 @@ public LocalDateTime getDateInSpecificTimeZone(String dateInString, String zone)
      * OffsetDateTime советуют использовать при записи даты в базу данных.
      */
     public OffsetDateTime offsetDateTime(LocalDateTime localTime) {
-        return OffsetDateTime.now();
+        return OffsetDateTime.of(localTime, ZoneOffset.of(UA_OFFSET));
     }
 
     /**
      * Дана строка в виде "yyyymmdd".
      * Необходимо вернуть Optional даты в LocalDate формате
      */
     public Optional<LocalDate> parseDate(String date) {
-        return Optional.empty();
+        try {
+            return Optional.of(LocalDate.parse(date, DateTimeFormatter.BASIC_ISO_DATE));
+        } catch (Exception e) {
+            return Optional.empty();
```

Comment:
same

Reply:
If i wont use "try-catch" as "if-else" here i definitely should implement new method "isDateValid()" and use "try-catch" as "if-else" in it to check validity of a date.